core/node/node_framework/src/implementations/layers/l1_gas.rs

```diff
@@ -75,8 +84,18 @@ impl WiringLayer for SequencerL1GasLayer {
         .context("GasAdjuster::new()")?;
         let gas_adjuster = Arc::new(adjuster);
 
+        let pool_resource = context.get_resource::<PoolResource<ReplicaPool>>().await?;
+        let replica_pool = pool_resource.get().await?;
+
+        let base_token_adjuster = MainNodeBaseTokenAdjuster::new(
+            replica_pool.clone(),
+            self.base_token_adjuster_config,
+            self.base_token_l1_address,
+        );
```

Comment:
I thought `Arc<dyn BaseTokenAdjuster>` would be a resource.
Then here we may fetch this resource, and if it's not provided, use a `NoOpBaseTokenAdjuster` which will always return 1.
Then separately, we will have `CustomBaseTokenAdjusterLayer` which will provide a resource and spawn the required maintenance task.

This way if we add `CustomBaseTokenAdjusterLayer`, gas prices will be adjusted. If we don't add, the system will work just fine for ETH.